shared/people/wot-task/index.tsx

```diff
@@ -0,0 +1,109 @@
+import PeopleItem, {TaskButton} from '../item'
+import * as Kb from '../../common-adapters'
+import * as React from 'react'
+import * as Styles from '../../styles'
+import {WotStatusType} from '../../constants/types/rpc-gen'
+
+type Props = {
+  userForIcon: string
+  otherUser: string
+  voucher: string
+  key: string
+  vouchee: string
+  wotStatus: WotStatusType
+  badged: boolean
+  onClickUser: (username: string) => void
+  onDismiss: (voucher: string, vouchee: string) => void
+}
+
+function makeButtons(props: Props) {
+  let handleButton = {} as TaskButton
+  switch (props.wotStatus) {
+    case WotStatusType.proposed:
+      handleButton = {
+        label: 'Review claim',
+        onClick: () => props.onClickUser(props.vouchee),
+      }
+      break
+    case WotStatusType.accepted:
+      handleButton = {
+        label: 'View claim',
+        onClick: () => props.onClickUser(props.vouchee),
+      }
+      break
+    case WotStatusType.rejected:
+      handleButton = {
+        label: 'Edit claim',
+        onClick: () => props.onClickUser(props.vouchee),
+      }
+      break
+  }
+  return [
+    handleButton,
+    {
+      label: 'Dismiss',
+      mode: 'Secondary',
+      onClick: () => props.onDismiss(props.voucher, props.vouchee),
+    },
+  ] as Array<TaskButton>
+}
+
+const makeMessage = (props: Props) => {
+  const connectedUsernamesProps = {
+    colorFollowing: true,
+    inline: true,
+    joinerStyle: {
+      fontWeight: 'normal',
+    },
+    onUsernameClicked: 'profile',
+    type: 'BodyBold',
+    underline: true,
+  } as const
+  const voucherComponent = (
+    <Kb.ConnectedUsernames
+      {...connectedUsernamesProps}
+      usernames={props.voucher}
+      onUsernameClicked={props.onClickUser}
+    />
+  )
+  const voucheeComponent = (
+    <Kb.ConnectedUsernames
+      {...connectedUsernamesProps}
+      usernames={props.vouchee}
+      onUsernameClicked={props.onClickUser}
+    />
+  )
+  switch (props.wotStatus) {
+    case WotStatusType.proposed:
+      return <Kb.Text type="Body">{voucherComponent} submitted an entry to your web of trust.</Kb.Text>
+    case WotStatusType.accepted:
+      return <Kb.Text type="Body">{voucheeComponent} accepted your entry into their web of trust.</Kb.Text>
+    case WotStatusType.rejected:
+      return <Kb.Text type="Body">{voucheeComponent} rejected your entry into their web of trust.</Kb.Text>
+    default:
+      return <Kb.Text type="Body">unknown.</Kb.Text>
+  }
+}
```

Comment:
Consider rewriting this function to return an object instead. All of the `ConnectedUsernames` and `Text` components can be rendered in `WotTask`

```js
{
    message: string,
    usernames: props.voucher | props.vouchee
}
```

Reply:
the reason i did it this way is that we can't guarantee the ConnectedUsername component will always be exactly one of voucher/vouchee and always at the beginning of the message. it's actually quite expected (i.e. in the designs) that we'll want to tell you that someone you follow has an accepted attestation from someone else you follow.

Reply:
Got it, thanks for the explanation. This makes more sense.

I suppose the follow up suggestion would be to refactor the common elements out `Kb.Text` for example. `onUsernameClicked` can move to the common props.

Optionally you can combine `const voucheeComponent` and `const voucherComponent` together. The only prop different is `usernames`.

Otherwise looks good.